src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1075,12 +1075,26 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
               return areStringsEqual(left.getString(), right.getString()).not();
           }
         }
+        switch (op) {
+          case Token.GE:
+          case Token.LE:
+          case Token.GT:
+          case Token.LT:
+            return compareAsNumbers(op, left, right, useTypes);
```

Comment:
I don't think this is a correct optimization. Consider, for example: `'a'  < 'b'` vs. `+'a' < +'b'`